### PR TITLE
Connect aug filters

### LINKED ISSUE #92: Create connection between Augmentations and Filters
Right now, filters and augmentations work on original images.
But if user wants to apply filters on augmentated images, which is not implemented.
Now implement code for 
1. Create connection between filters.py and Augmentations.py in such a way that user should be able to apply filters on augmented images(those are stored in ProcessedData folder)

### PR DESCRIPTION
**Pull request by rubyruins**
# Description:

Creates connection between filters.py and Augmentations.py in such a way that user should be able to apply filters on augmented images and vice versa

Fixes #92 

## Type of change:

- [x] New feature (non-breaking change which adds functionality)
 
# Screenshots / Video:

Running the following code to apply augmentations on filters
```
filters = ["median", "laplacian"]
flow.applyFilters( filters )
operations = {'flipped': 'horizontal', 'rotated': 90, 'apply_on_filters': ['laplacian']}
flow.applyAugmentation( operations )
```
Rotatedlaplacianapple_2 and Rotatedlaplacianapple_3

<p float="left">

</p>

Running the following code to apply filters on augmentations
```
operations = {'flipped': 'horizontal', 'rotated': 90}
flow.applyAugmentation( operations )
filters = ["median", "laplacian", {'apply_on_augmentation': ['rotated']}]
flow.applyFilters( filters )
```
MedianRotatedorange_2 and MedianRotatedorange_3

<p float="left">

</p>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -118,18 +118,30 @@ Flow2ML for Python can be installed from Source,Pypi and Docker container instal
...
-    filters = ["median", "laplacian", "gaussian", "sobelx", "sobely","bilateral"]
+    filters = ["median", "laplacian", "gaussian", "sobelx", "sobely", "bilateral"]
 
     # Apply The Filters
     flow.applyFilters( filters )
 
     # Define The augmentation operations to be used
    
-    operations = {'flip': 'horizontal', 'rotate': 90, 'shear': {'x_axis': 5, 'y_axis': 15}, 'crop': [50, 100, 50, 100], 'scale': 0.1, 'zoom': 2, 'Hist_Equal':False, 'greyscale': True, 'CLAHE':False, 'invert':False, 'erode':False, 'dilate':False, 'open':False, 'close':False,'threshold':{'type':'adaptive','thresh_val':0},'color-space':{'input':'BGR','output':'BGR'}}
+    operations = {'flipped': 'horizontal', 'rotated': 90, 'sheared': {'x_axis': 5, 'y_axis': 15}, 'cropped': [50, 100, 50, 100], 'scaled': 0.1, 'zoomed': 2, 'histogramequalised':False, 'greyscale': True, 'CLAHE':False, 'inverted':False, 'eroded':False, 'dilated':False, 'opened':False, 'closed':False,'thresholded':{'type':adaptive,'thresh_val':0}}
```
Comment (chebroluharika) [inline README.md:128]: Why did you remove color-space?

Comment (rubyruins) [reply README.md:128]: Sorry. Actually there was a merge conflict in readme so this line got removed. I'll add it back

File: flow2ml/Flow.py [hunk 2]
```diff
@@ -102,57 +117,66 @@ def applyAugmentation(self,operations):
...
 
-        if operation == "close":
-          self.applyClose(path)
+          if operation == "closed":
+            self.applyClose(path)
 
-        if operation == "threshold":
-          self.applyThreshold(path)
+          if operation == "thresholded":
+            self.applyThreshold(path)
```
Comment (chebroluharika) [inline flow2ml/Flow.py:179]: Change formatting of line 181 and 182 too

File: README.md (hunk 1 above)
Comment (yvkrishna) [reply README.md]: @rubyruins, can you change the cropping to random crop instead of a list.

Generally we would use random cropping.

It's okay to specify the points, but try to check it with random crop also.

Comment (rubyruins) [reply README.md]: @yvkrishna random crop is also implemented in data_augmentation.py. This is just some sample code in the readme.

File: flow2ml/Flow.py (hunk 2 above)
Comment (rubyruins) [reply flow2ml/Flow.py:179]: Done

**Review by chebroluharika — APPROVED** [review]
